examples/go/tagging/handler/handler.go

```diff
@@ -0,0 +1,244 @@
+	log.Println(message)
+
+	return handler.Response{
+		Body:       []byte(message),
+		StatusCode: http.StatusOK,
+	}, nil
+}
+
+// vsConnect connects to vSphere govmomi API using information from vcconfig.toml.
+func vsConnect(ctx context.Context, cfg *vcConfig) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if client == nil {
+		u := url.URL{
+			Scheme: "https",
+			Host:   cfg.VCenter.Server,
+			Path:   "sdk",
+		}
+		u.User = url.UserPassword(cfg.VCenter.User, cfg.VCenter.Password)
+		insecure := cfg.VCenter.Insecure
+
+		if debug() {
+			log.Println("connect to vSphere")
+		}
+
+		c, err := NewClient(ctx, u, insecure)
+		if err != nil {
+			return fmt.Errorf("connection to vSphere API failed: %w", err)
+		}
+
+		// Set global variable to persist connection.
+		client = c
+	}
+
+	return nil
+}
+
+func loadTomlCfg(path string) (*vcConfig, error) {
+	var cfg vcConfig
+
+	secret, err := toml.LoadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load vcconfig.toml: %w", err)
+	}
+
+	err = secret.Unmarshal(&cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal vcconfig.toml: %w", err)
+	}
+
+	err = validateConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("insufficient information in vcconfig.toml: %w", err)
+	}
+
+	return &cfg, nil
+}
+
+// ValidateConfig ensures the bare minimum of information is in the config file.
+func validateConfig(cfg vcConfig) error {
+	reqFields := map[string]string{
+		"vcenter server":   cfg.VCenter.Server,
+		"vcenter user":     cfg.VCenter.User,
+		"vcenter password": cfg.VCenter.Password,
+		"tag URN":          cfg.Tag.URN,
+		"tag action":       cfg.Tag.Action,
+	}
+
+	// Multiple fields may be missing, but err on the first encountered.
+	for k, v := range reqFields {
+		if v == "" {
+			return errors.New("required field(s) missing, including " + k)
+		}
+	}
+
+	return nil
+}
+
+// Debug determines verbose logging
+func debug() bool {
+	verbose := os.Getenv("write_debug")
+
+	if verbose == "true" {
+		return true
+	}
+
+	return false
+}
+
+func parseEventMoRef(req []byte) (*types.ManagedObjectReference, error) {
+	var event incoming
+	var moRef types.ManagedObjectReference
+
+	err := json.Unmarshal(req, &event)
+	if err != nil {
+		return nil, fmt.Errorf("parsing of request failed: %w", err)
+	}
+
+	if event.Data.Vm == nil || event.Data.Vm.Vm.Value == "" {
+		return nil, errors.New("empty managed reference object")
+	}
+
+	// Fill information in the request into a govmomi type.
+	moRef.Type = event.Data.Vm.Vm.Type
+	moRef.Value = event.Data.Vm.Vm.Value
+
+	return &moRef, nil
+}
+
+func handleSignal(ctx context.Context) {
+	var sigCh = make(chan os.Signal)
+
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
+
+	s := <-sigCh
+	verbose := debug()
+
+	if verbose {
+		log.Printf("got signal: %v, log out of vSphere", s)
+	}
+
+	err := client.logout(ctx)
+	if verbose {
+		if err != nil {
+			log.Printf("vSphere logout failed: %v", err)
+		} else {
```

Comment:
nit: instead of  `else` logic you can align the "happy path" (line of sight) with this pattern:

```go
	err := client.logout(ctx)
	if verbose {
		if err != nil {
			log.Printf("vSphere logout failed: %v", err)
			return
		}
		log.Println("logged out of govmomi and rest APIs")
	}
```

Reply:
I keep forgetting returning nothing is allowed. Good idea!